plugins/module_utils/nd_network_resources.py

```diff
@@ -0,0 +1,151 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, Gaspard Micol (@gmicol) <gmicol@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+from copy import deepcopy
+from functools import reduce
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, issubset, sanitize
+
+
+# TODO: Maybe make it an all new module class with request, exit_json, fail_json, etc...
+# TODO: Add description to functions and clean the code
+# TODO: Add exception cases
+# NOTE: ONLY works for new API endpoints introduced in ND v4.1.0 and later
+class NDNetworkResourceModule(NDModule):
```

Comment:
I like the approach of the class with generic methods that handle actions like merge etc

Reply:
+1, but if there are common lines can we move this to a central location such that all the other modules can utilize this?